projects/components/src/greeting-label/greeting-label.component.ts

```diff
@@ -0,0 +1,39 @@
+import { ChangeDetectionStrategy, Component, Input, OnChanges } from '@angular/core';
+import { TypedSimpleChanges } from '@hypertrace/common';
+
+@Component({
+  selector: 'htc-greeting-label',
+  styleUrls: ['./greeting-label.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: ` <div class="greeting-label">{{ this.greetingMessage }}</div>`
+})
+export class GreetingLabelComponent implements OnChanges {
+  @Input()
+  public readonly suffixLabel: string = '';
+
+  public greetingMessage: string = '';
+
+  public ngOnChanges(changes: TypedSimpleChanges<this>): void {
+    if (changes.suffixLabel) {
+      this.setGreetingMessage();
+    }
+  }
+
+  private setGreetingMessage(): void {
+    this.greetingMessage = `${this.getGreeting()}${this.suffixLabel}`;
+  }
+
+  private getGreeting(): string {
+    const hour = new Date().getHours();
+
+    if (hour < 12) {
+      return 'Good Morning';
```

Comment:
Some nitpicky suggestions at the time windows:
Morning 4AM-12PM Afternoon 12PM-5PM Evening otherwise (5PM-4AM)

Reply:
I have to send another PR to include this widget in home dash. Will make these changes along with that.